File: hyacinth/src/main/java/com/github/saturn_xiv/palm/hyacinth/repositories/LogRepository.java
package com.github.saturn_xiv.palm.hyacinth.repositories;

import java.util.List;

import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import com.github.saturn_xiv.palm.hyacinth.models.Log;

@Repository("palm.hyacinth.log-repository")
public interface LogRepository extends CrudRepository<Log, Integer> {
    @Query("SELECT i FROM Log i WHERE i.timeTaking > :value ORDER BY i.createdAt DESC")
    List<Log> findTimeTakingLagerThan(@Param("value") int value);
}

File: dahlia/src/main/kotlin/com/github/saturn_siv/palm/dahlia/plugins/bookkeeper/repositories/LogRepository.kt
package com.github.saturn_siv.palm.dahlia.plugins.bookkeeper.repositories

import com.github.saturn_siv.palm.dahlia.plugins.bookkeeper.models.Log
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository("dahlia.bookkeeper.r.log.jpa")
interface LogRepository : CrudRepository<Log, Int> {
}